src/apps/frontend/services/auth.service.ts

```diff
@@ -0,0 +1,21 @@
+import { AccessToken, ServiceError, ServiceResponse } from '../types';
+
+import APIService from './api.service';
+
+export default class AuthService extends APIService {
+  login = async (
+    username: string,
+    password: string,
+  ): Promise<ServiceResponse<AccessToken>> => {
```

Comment:
You mean `Promise<AsyncResponse<AccessToken, ApiError>>`? 
I can do this by doing `class ServiceResponse implements AsyncResponse` but then `AsyncResponse` has a `status` property. The service layer won't handle the status; instead, it will be managed by the `useAsync` hook, which sets it to values like `IDLE, PENDING, SUCCESS, or ERROR`."

Reply:
> The service layer won't handle the status

Should it?

Reply:
IMO No, because even if service change the status from Pending to Sucess or Error, UI won't get updated. Also then `useAsync` would be of no use as it cant be used inside a service.

Reply:
The service layer should not have dependencies or concerns related to the UI layer.

Reply:
I think `status` is not a concern of UI layer, rather its an inherent property of any async operation.

Reply:
But then it should be limited to SUCCESS or ERROR only. And there will be a middle layer(hook) which will expose `loading: boolean` to UI layer